core/src/test/java/org/mule/management/stats/ComponentStatisticsTestCase.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.management.stats;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+
+/**
+ * Validates some basic assumptions about the ComponentStatistics class
+ * behavior.
+ */
+public class ComponentStatisticsTestCase extends AbstractMuleTestCase {
+
+    /* This class was added to address MULE-6417 */
+    
+    private String originalStatIntervalTime;
+    
+    @Before
+    public void setUp() {
+        // ComponentStatistics reads statIntervalTime from the environment.
+        // We will want to control it.
+        originalStatIntervalTime = System.getProperty("statIntervalTime");
+        System.clearProperty("statIntervalTime");
+    }
+    
+    @After
+    public void tearDown() {
+        if (originalStatIntervalTime != null) {
+            System.setProperty("statIntervalTime", originalStatIntervalTime);
+        }
+    }
+    
+    private void assertValues(ComponentStatistics stats,
+            long numEvents, long totalTime, long avgTime, 
+            long maxTime, long minTime) {
+        assertThat("getExecutedEvents", stats.getExecutedEvents(), equalTo(numEvents));
+        assertThat("getTotalExecutionTime", stats.getTotalExecutionTime(), equalTo(totalTime));
+        assertThat("getAverageExecutionTime", stats.getAverageExecutionTime(), equalTo(avgTime));
+        assertThat("getMaxExecutionTime", stats.getMaxExecutionTime(), equalTo(maxTime));
+        assertThat("getMinExecutionTime", stats.getMinExecutionTime(), equalTo(minTime)); 
+    }
+    
+    @Test
+    public void verifyStatDefaults() {
+        ComponentStatistics stats = new ComponentStatistics();
+        // num, total, avg, max, min
+        assertValues(stats, 0L, 0L, 0L, 0L, 0L);
+        assertThat(stats.isEnabled(), equalTo(false));
+    }
+    
+    @Test
+    public void processSingleEvent() {
+        ComponentStatistics stats = new ComponentStatistics();
+        stats.addExecutionTime(100L);
+        // num, total, avg, max, min
+        assertValues(stats, 1L, 100L, 100L, 100L, 100L);
+    }
+    
+    @Test
+    public void processSingleBranchEvent() {
+        ComponentStatistics stats = new ComponentStatistics();
+        stats.addExecutionBranchTime(true /*first*/, 25L, 25L);
```

Comment:
Can this inline comments be changed for something more explanatory? If not, just remove them.